src/main.ts

```diff
@@ -169,3 +216,25 @@ app.on('activate', () => {
     createWindow()
   }
 })
+
+if (!app.isDefaultProtocolClient(pkg.protocolSchemes.native)) {
+  // Define custom protocol handler. Deep linking works on packaged versions of the application!
+  app.setAsDefaultProtocolClient(pkg.protocolSchemes.native)
+}
+
+app.on('will-finish-launching', function () {
+  // Protocol handler for osx
+  app.on('open-url', function (event, url) {
+    event.preventDefault()
+    deeplinkingUrl = url
+    if (mainWindow) {
+      redirectDeepLinkingUrl(deeplinkingUrl.toString(), mainWindow)
+    }
+  })
+})
+
+ipcMain.on('app-ready', () => {
+  if (mainWindow && deeplinkingUrl) {
+    redirectDeepLinkingUrl(deeplinkingUrl.toString(), mainWindow)
+  }
+})
```

Comment:
I think we need to start putting stuff from the main file to a different folder, because `main` becomes a very large and unstable blob of code. Can we move this, let's say, to the `features/deepLinking` ? I'm not sure how right is that, I'll let you decide where it should go, but IMO having it in features is okay.

Also, having these function separately will allow to test them as well